### PR TITLE
fix: kiuwan bugs

### PR DESCRIPTION
**Pull request by mvarlic**
* fix multiples kiuwan bugs

image
image
image

image
image
image
image
image
image

### CODE REVIEW

**Review by victormendoza96 — APPROVED** [review]

File: webpay/webpay.php [hunk 1]
```diff
@@ -48,7 +49,7 @@ public function __construct()
         try {
             $this->log = TbkFactory::createLogger();
         } catch (Exception $e) {
-            print_r($e);
+            throw new EcommerceException($e->getMessage(), 0, $e);
```
Comment (Matiasnickolas) [inline webpay/webpay.php:52]: replace this magic number '0'

File: webpay/src/Utils/TransbankSdkWebpay.php [hunk 2]
```diff
@@ -108,11 +102,8 @@ public function commitTransaction($tokenWs)
 
             return $this->transaction->commit($tokenWs);
         } catch (TransactionCommitException $e) {
-            $result = [
-                'error'  => 'Error al confirmar la transacción',
-                'detail' => $e->getMessage(),
-            ];
-            $this->log->logError(json_encode($result));
+            $errorMessage = "Error confirmando la transacción para => tokenWs: {$tokenWs}, error: {$e->getMessage()}";
+            throw new EcommerceException($errorMessage, 0, $e);
```
Comment (Matiasnickolas) [inline webpay/src/Utils/TransbankSdkWebpay.php:109]: replace this magic number

File: webpay/src/Utils/TransbankSdkWebpay.php [hunk 3]
```diff
@@ -71,18 +71,12 @@ public function createTransaction($amount, $sessionId, $buyOrder, $returnUrl)
...
+                throw new EcommerceException($errorMessage);
             }
         } catch (Exception $e) {
-            $result = [
-                'error'  => 'Error al crear la transacción',
-                'detail' => $e->getMessage(),
-            ];
-            $this->log->logError('createTransaction.error: '.json_encode($result));
+            $errorMessage = "Error creando la transacción para => buyOrder: {$buyOrder}, amount: {$amount}, error: {$e->getMessage()}";
+            throw new EcommerceException($errorMessage, 0, $e);
```
Comment (Matiasnickolas) [inline webpay/src/Utils/TransbankSdkWebpay.php:82]: replace this magic number

File: webpay/src/Utils/TransbankSdkOneclick.php [hunk 4]
```diff
@@ -72,23 +72,20 @@ public function startInscription($userName, $email, $returnUrl)
...
             }
         } catch (Exception $e) {
-            $result = [
-                'error'  => 'Error al iniciar la inscripción',
-                'detail' => $e->getMessage().$returnUrl,
-            ];
-            $this->log->logError(json_encode($result));
+            $errorMessage = "Error al iniciar la inscripción para => userName: {$userName}, email: {$email}, error: {$e->getMessage()}";
+            $this->log->logError($errorMessage);
+            throw new EcommerceException($errorMessage, 0, $e);
```
Comment (Matiasnickolas) [inline webpay/src/Utils/TransbankSdkOneclick.php:90]: replace this magic number

File: webpay/src/Utils/TransbankSdkOneclick.php [hunk 5]
```diff
@@ -102,28 +99,22 @@ public function startInscription($userName, $email, $returnUrl)
...
             return $resp;
         } catch (Exception $e) {
-            $result = [
-                'error'  => 'Error al confirmar la inscripción',
-                'detail' => $e->getMessage(),
-            ];
-            $this->log->logError(json_encode($result));
+            $errorMessage = "Error al confirmar la inscripción para => userName: {$userName}, email: {$email}, error: {$e->getMessage()}";
+            $this->log->logError($errorMessage);
+            throw new EcommerceException($errorMessage, 0, $e);
```
Comment (Matiasnickolas) [inline webpay/src/Utils/TransbankSdkOneclick.php:119]: replace this magic number

File: webpay/src/Utils/TransbankSdkOneclick.php [hunk 6]
```diff
@@ -155,20 +145,15 @@ public function authorizeTransaction($username, $tbkUser, $parentBuyOrder, $chil
...
             return $resp;
         } catch (Exception $e) {
-            $result = [
-                'error'  => 'Error al autorizar el pago',
-                'detail' => $e->getMessage(),
-            ];
-            $this->log->logError(json_encode($result));
+            $errorMessage = "Error al autorizar el pago para => userName: {$username}, buyOrder: {$parentBuyOrder}, error: {$e->getMessage()}";
+            $this->log->logError($errorMessage);
+            throw new EcommerceException($errorMessage, 0, $e);
```
Comment (Matiasnickolas) [inline webpay/src/Utils/TransbankSdkOneclick.php:159]: replace this magic number

Comment (mvarlic) [reply webpay/src/Utils/TransbankSdkOneclick.php]: added constructor with default code exception

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    public function logOneclickInstallConfigLoadDefaultPorIncompleta(){
+        $this->logInfo('Configuración por defecto de ONECLICK se cargo de forma correcta porque los valores de producción estan incompletos');
+    }
+
+    public function logOneclickConfigError(){
+        $this->logError('Configuración de ONECLICK incorrecta, revise los valores');
+    }
+
+    public function logOneclickPaymentIniciando(){
+        if($this->getDebugActive()==1){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:30]: add constant for this magic number

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            $this->logInfo('B.1. Iniciando medio de pago Oneclick');
+        }
+    }
+
+    public function logOneclickPaymentAntesObtenerInscripcion($inscriptionId, $cartId, $amount){
+        if($this->getDebugActive()==1){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:36]: add constant for this magic number

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+    public function logOneclickInstallConfigLoadDefaultPorIncompleta(){
+        $this->logInfo('Configuración por defecto de ONECLICK se cargo de forma correcta porque los valores de producción estan incompletos');
+    }
+
+    public function logOneclickConfigError(){
+        $this->logError('Configuración de ONECLICK incorrecta, revise los valores');
+    }
+
+    public function logOneclickPaymentIniciando(){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:29]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    public function logOneclickInstallConfigLoad($oneclickMallCommerceCode, $oneclickChildCommerceCode, $oneclickDefaultOrderStateIdAfterPayment){
+        $this->logInfo('Configuración de ONECLICK se cargo de forma correcta');
+        $this->logInfo('oneclickMallCommerceCode: '.$oneclickMallCommerceCode.', oneclickChildCommerceCode: '.$oneclickChildCommerceCode.', oneclickDefaultOrderStateIdAfterPayment: '.$oneclickDefaultOrderStateIdAfterPayment);
+    }
+
+    public function logOneclickInstallConfigLoadDefault(){
+        $this->logInfo('Configuración por defecto de ONECLICK se cargo de forma correcta');
+    }
+
+    public function logOneclickInstallConfigLoadDefaultPorIncompleta(){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:20]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        }
+    }
+
+    public function logOneclickPaymentAntesObtenerInscripcion($inscriptionId, $cartId, $amount){
+        if($this->getDebugActive()==1){
+            $this->logInfo('B.2. Antes de obtener inscripción de la BD => inscriptionId: '.$inscriptionId.', cartId: '.$cartId.', amount: '.$amount);
+        }
+    }
+
+    public function logOneclickPaymentDespuesObtenerInscripcion($inscriptionId, $ins){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:42]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        if($this->getDebugActive()==1){
+            $this->logInfo('B.2. Despues de obtener inscripción de la BD => inscriptionId: '.$inscriptionId);
+            $this->logInfo(json_encode($ins));
+        }
+    }
+
+    public function logOneclickPaymentAntesCrearTxBd($inscriptionId, $transaction){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:49]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        if($this->getDebugActive()==1){
+            $this->logInfo('B.3. Preparando datos antes de crear la transacción en BD => inscriptionId: '.$inscriptionId);
+            $this->logInfo(json_encode($transaction));
+        }
+    }
+
+    public function logOneclickPaymentCrearTxBdError($inscriptionId, $transaction){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:57]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    public function logOneclickPaymentAntesAutorizarTx($username, $tbkToken, $parentBuyOrder, $childBuyOrder, $amount){
+        if($this->getDebugActive()==1){
+            $this->logInfo('B.4. Preparando datos antes de autorizar la transacción en Transbank');
+            $this->logInfo('username: '.$username.', tbkToken: '.$tbkToken.', parentBuyOrder: '.$parentBuyOrder.', childBuyOrder: '.$childBuyOrder.', amount: '.$amount);
+        }
+    }
+
+    public function logOneclickPaymentDespuesAutorizarTx($username, $tbkToken, $parentBuyOrder, $childBuyOrder, $amount, $result){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        $this->logInfo('B.5. Transacción con autorización en Transbank => username: '.$username.', tbkToken: '.$tbkToken.', parentBuyOrder: '.$parentBuyOrder.', childBuyOrder: '.$childBuyOrder.', amount: '.$amount);
+        $this->logInfo(json_encode($result));
+        if (!is_array($result) && $result->isApproved()){
+            $this->logInfo('***** AUTORIZADO POR TBK OK *****');
+            $this->logInfo('TRANSACCION VALIDADA POR TBK => username: '.$username.', tbkToken: '.$tbkToken.', parentBuyOrder: '.$parentBuyOrder.', childBuyOrder: '.$childBuyOrder.', amount: '.$amount);
+            $this->logInfo('SI NO SE ENCUENTRA VALIDACION POR PRESTASHOP DEBE ANULARSE');
+        }
+    }
+
+    public function logOneclickPaymentDespuesAutorizarTxError($parentBuyOrder, $childBuyOrder, $result){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:83]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithOneclickLog.php [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+        $this->logError('B.6. Transacción con autorización con error => parentBuyOrder: '.$parentBuyOrder.', childBuyOrder: '.$childBuyOrder);
+        $this->logError(json_encode($result));
+    }
+
+    public function logOneclickPaymentDespuesAutorizarRechazadoTxError($parentBuyOrder, $childBuyOrder, $result){
```
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithOneclickLog.php:89]: rename the method name, not use spanish

File: webpay/src/Helpers/InteractsWithWebpayLog.php
Comment (mastudillot) [inline webpay/src/Helpers/InteractsWithWebpayLog.php:1]: add constant for the magic numbers and rename methods with spanish names to use english

**Review by mastudillot — CHANGES_REQUESTED** [review]

**Comment by Matiasnickolas** [comment]
As recomendation, when a commit is related to correction of spacing, code formatting, missing quotes, etc. the commit token should be style instead refactor

File: webpay/shared/Helpers/ExceptionConstans.php [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Transbank\Plugin\Helpers;
+
+class ExceptionConstans
```
[the remark this answers is not in the document]
Comment (mvarlic) [reply webpay/shared/Helpers/ExceptionConstans.php]: its ready

**Review by Matiasnickolas — COMMENTED** [review]
GJ 🤙

**Review by Matiasnickolas — APPROVED** [review]

**Review by mastudillot — APPROVED** [review]
LGTM

**Pull request merged**